OptimizelySDK/Odp/RestApiManager.cs

```diff
@@ -0,0 +1,164 @@
+    /// </summary>
+    public class RestApiManager : IRestApiManager
+    {
+        private const string EVENT_SENDING_FAILURE_MESSAGE = "ODP event send failed";
+
+        /// <summary>
+        /// Error handler used to record errors
+        /// </summary>
+        private readonly IErrorHandler _errorHandler;
+
+        /// <summary>
+        /// Logger used to record messages that occur within the ODP client 
+        /// </summary>
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Http client used for handling requests and responses over HTTP
+        /// </summary>
+        private readonly HttpClient _httpClient;
+
+        /// <summary>
+        /// Manager for communicating with ODP's REST API endpoint
+        /// </summary>
+        /// <param name="logger">Collect and record events to log</param>
+        /// <param name="errorHandler">Handler to record exceptions</param>
+        /// <param name="httpClient">HttpClient to use to send ODP events</param>
+        public RestApiManager(ILogger logger = null, IErrorHandler errorHandler = null,
+            HttpClient httpClient = null
+        )
+        {
+            _logger = logger ?? new NoOpLogger();
+            _errorHandler = errorHandler ?? new NoOpErrorHandler();
+            _httpClient = httpClient ?? new HttpClient();
+        }
+
+        /// <summary>
+        /// Send events to ODP's RESTful API
+        /// </summary>
+        /// <param name="apiKey">ODP public key</param>
+        /// <param name="apiHost">Host of ODP endpoint</param>
+        /// <param name="events">ODP events to send</param>
+        /// <returns>Retry is true - if network or server error (5xx), otherwise false</returns>
+        public bool SendEvents(string apiKey, string apiHost, List<OdpEvent> events)
+        {
+            if (string.IsNullOrWhiteSpace(apiKey) || string.IsNullOrWhiteSpace(apiHost))
+            {
+                _logger.Log(LogLevel.ERROR,
+                    $"{EVENT_SENDING_FAILURE_MESSAGE} (Parameters apiKey or apiHost invalid)");
+                return false;
+            }
+
+            if (events.Count == 0)
+            {
+                _logger.Log(LogLevel.ERROR, $"{EVENT_SENDING_FAILURE_MESSAGE} (no events)");
+                return false;
+            }
+
+            var endpoint = $"{apiHost}/v3/events";
+            var data = JsonConvert.SerializeObject(events);
+            var shouldRetry = false;
+
+            HttpResponseMessage response = default;
+            try
+            {
+                response = SendEventsAsync(apiKey, endpoint, data).GetAwaiter().GetResult();
+            }
+            catch (Exception ex)
+            {
+                _errorHandler.HandleError(ex);
+                _logger.Log(LogLevel.ERROR, $"{EVENT_SENDING_FAILURE_MESSAGE} (network error)");
+                shouldRetry = true;
+            }
+
+            var responseStatusCode = response == null ? 0 : (int)response.StatusCode;
```

Comment:
same ensurestatuscode

Reply:
Looking at the [Swift reference,](https://github.com/optimizely/swift-sdk/pull/455/files#diff-e050cd60c809ae12d70b80d9474c73b63d5891d6afa08b66b7e900ec13079f4bR89) for the REST API Event Manager, it seems like we're not going to attempt a retry only for server-side errors >= 500.

Perhaps discuss?